### ISSUE TITLE
hcl: Crash when no 'default' given in main.tf.json variable

### ISSUE THREAD

**Issue by adamdecaf** [hashicorp/terraform#8580]
### Terraform Version
Run `terraform -v` to show the version. If you are not running the latest version of Terraform, please upgrade because your issue may have already been fixed.

```
adam@Planet-X -- ~:  terraform version
Terraform v0.7.1

Your version of Terraform is out of date! The latest version
is 0.7.2. You can update by downloading from www.terraform.io
```

Terraform [v0.7.2](https://github.com/hashicorp/terraform/blob/v0.7.2/vendor/github.com/hashicorp/hcl/hcl/ast/ast.go#L58) and [v0.7.1](https://github.com/hashicorp/terraform/blob/v0.7.1/vendor/github.com/hashicorp/hcl/hcl/ast/ast.go#L58) have the same vendored version of hashicorp/hcl. 

### Affected Resource(s)
Please list the resources as a list, for example:
- hcl parsing

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
{
  "module": {
    "a": {
      "source": "./a/",
      "foo": "${element(split(\",\", var.hosts), 0)}"
    }
  },
  "variable": {
    "hosts": "web1.example.com,web2.example.com"
  }
}
```

I've created [failing and passing](https://github.com/adamdecaf/tf-crash-2016-08-31/tree/master/json-crash) main.tf.json examples. (As well as an .hcl example that passes.) 

### Debug Output
Please provider a link to a GitHub Gist containing the complete debug output: https://www.terraform.io/docs/internals/debugging.html. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

That's available on my repo: https://github.com/adamdecaf/tf-crash-2016-08-31/blob/master/json-crash/debug.log

### Panic Output
If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`.

I've included the crash log on my repo: https://github.com/adamdecaf/tf-crash-2016-08-31/blob/master/json-crash/crash.log

### Expected Behavior
What should have happened?

I'd expect terraform to not crash, but perhaps give a message saying the json is invalid. (Missing `default` object key.)

### Actual Behavior
What actually happened?

Terraform crashed. :( 

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform plan`

### Important Factoids
Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? Custom version of OpenStack? Tight ACLs?

None.

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:
- The attached repo above.

**Note** I did try my hand at introducing some guards around [nil in hashicorp/hcl](https://github.com/hashicorp/hcl/compare/master...adamdecaf:json-missing-defaults-crash) but have yet to produce a failing test case first. I assume I'd need to setup a more specific parse and object filter test to uncover this.

**Comment by mitchellh** [comment]
I can't get this to crash with mater, I used your exact config with module "a" being:

```
variable "foo" {}
```

Plan, apply all work. 

Is there something I'm missing?

**Comment by adamdecaf** [comment]
I tried master as well and I'm not getting a crash. This seems to be fixed in 0.7.3 then. Cool!
